lib/cocoapods/installer/analyzer.rb

```diff
@@ -823,6 +838,31 @@ def determine_platform(specs, target_definitions, build_type)
         Platform.new(platform_name, deployment_target)
       end
 
+      # Determines the Swift version for the given spec within the target definition. If the pod author has provided a
+      # set of Swift versions supported by their pod then the max Swift version is chosen, unless the target definitions
+      # specify explicit requirements for supported Swift versions. Otherwise the Swift version is derived by the
+      # target definitions that integrate this pod.
+      #
+      # @param [Specification] spec
+      #        the specs to inspect and determine what Swift version to use.
+      #
+      # @param [Array<TargetDefinition>] target_definitions
+      #        the target definition the spec is part of.
+      #
+      # @return [String, nil] the computed Swift version or `nil` if the Swift version could not be determined.
+      #
+      def determine_swift_version(spec, target_definitions)
```

Comment:
this was basically moved from `pod_target.rb` because we need it here during the variants calculation.